### PR TITLE
shutdown thread pool

### PR DESCRIPTION
**Pull request by DerekYRC**
## PR Type

Other

## Describe what this PR does for and how you did.

shutdown thread pool

## Checklist

- [x] Add copyright holder at the beginning of .class file if it is new.
- [x] Add information of this PR to CHANGELOG.md in root of project.
- [x] All junit tests passing.
- [x] Coverage from `Codecov Report` should not decrease.

## Checklist (Optional)

- [x] Will Pull Request to branch of 2020.0 and 2021.0.
- [x] Add documentation in javadoc in code or comment below the PR if necessary.
- [x] Add your name as @author to the beginning of .class file.

### CODE REVIEW

File: spring-cloud-starter-tencent-polaris-discovery/src/main/java/com/tencent/cloud/polaris/registry/PolarisServiceRegistry.java [hunk 1]
```diff
@@ -216,4 +217,9 @@ public void heartbeat(InstanceHeartbeatRequest heartbeatRequest) {
 			}
 		}, polarisDiscoveryProperties.getHeartbeatInterval(), polarisDiscoveryProperties.getHeartbeatInterval(), MILLISECONDS);
 	}
+
+	@Override
+	public void destroy() throws Exception {
+		heartbeatExecutor.shutdown();
```
Comment (lepdou) [inline spring-cloud-starter-tencent-polaris-discovery/src/main/java/com/tencent/cloud/polaris/registry/PolarisServiceRegistry.java:223]: 这里应该要判断一下 heatbeatExecutor 是否为null。因为创建的时候，有开关控制：

````
if (polarisDiscoveryProperties.isHeartbeatEnabled()) {
			this.heartbeatExecutor = Executors.newSingleThreadScheduledExecutor(
					new NamedThreadFactory("spring-cloud-heartbeat"));
		}
````

Comment (DerekYRC) [reply spring-cloud-starter-tencent-polaris-discovery/src/main/java/com/tencent/cloud/polaris/registry/PolarisServiceRegistry.java]: Done.

**Comment by SkyeBeFreeman** [comment]
This executor is closed when deregister method is called. Is it necessary to keep only one close method?

**Comment by DerekYRC** [comment]
> This executor is closed when deregister method is called. Is it necessary to keep only one close method?

没有注意到heartbeatExecutor被关闭的代码，我关闭PR了哈

**Pull request closed without merging**